ksp_plugin_test/flight_plan_optimizer_test.cpp

```diff
@@ -344,5 +373,191 @@ TEST_F(FlightPlanOptimizerTest, DISABLED_Combined) {
+
+  Instant const epoch_;
+  TestNavigationFrame const navigation_frame_;
+  NavigationManœuvre::Burn const burn_;
+
+  DiscreteTrajectory<Barycentric> root_;
+  std::unique_ptr<FlightPlan> flight_plan_;
+  std::unique_ptr<FlightPlanOptimizer> optimizer_;
+  std::unique_ptr<FlightPlanOptimizer::Metric> metric_;
+};
+
+SolarSystem<Barycentric> const MetricTest::solar_system_1950_(
+    SOLUTION_DIR / "astronomy" / "sol_gravity_model.proto.txt",
+    SOLUTION_DIR / "astronomy" /
+        "sol_initial_state_jd_2433282_500000000.proto.txt",
+    /*ignore_frame=*/true);
+
+not_null<std::unique_ptr<Ephemeris<Barycentric>>> const MetricTest::ephemeris_(
+    solar_system_1950_.MakeEphemeris(
+        /*accuracy_parameters=*/{/*fitting_tolerance=*/5 * Milli(Metre),
+                                 /*geopotential_tolerance=*/0x1p-24},
+        Ephemeris<Barycentric>::FixedStepParameters(
+            SymmetricLinearMultistepIntegrator<
+                QuinlanTremaine1990Order12,
+                Ephemeris<Barycentric>::NewtonianMotionEquation>(),
+            /*step=*/10 * Minute)));
+
+not_null<RotatingBody<Barycentric> const*> const MetricTest::earth_body_(
+    solar_system_1950_.rotating_body(*ephemeris_, "Earth"));
+
+not_null<std::unique_ptr<Celestial>> const MetricTest::earth_celestial_ = []() {
+  auto celestial = std::make_unique<Celestial>(earth_body_);
+  celestial->set_trajectory(ephemeris_->trajectory(earth_body_));
+  return celestial;
+}();
+
+TEST_P(MetricTest, Positive) {
+  std::mt19937_64 random(42);
+  std::uniform_real_distribution<double> coordinate(-100, 100);
+  for (int i = 0; i < 100; ++i) {
+    FlightPlanOptimizer::HomogeneousArgument const argument(
+        std::array{coordinate(random),
+                   coordinate(random),
+                   coordinate(random),
+                   coordinate(random)});
+    EXPECT_LE(0, metric_->Evaluate(argument));
+  }
+}
+
+TEST_P(MetricTest, Gradient) {
+  std::mt19937_64 random(42);
+  std::uniform_real_distribution<double> coordinate(-100, 100);
+  std::uniform_real_distribution<double> displacement(-1, 1);
+  for (int i = 0; i < 100; ++i) {
+    FlightPlanOptimizer::HomogeneousArgument const argument(
+        std::array{coordinate(random),
+                   coordinate(random),
+                   coordinate(random),
+                   coordinate(random)});
+    for (int j = 0; j < 10; ++j) {
+      FlightPlanOptimizer::HomogeneousArgument const Δargument(
+          std::array{displacement(random),
+                     displacement(random),
+                     displacement(random),
+                     displacement(random)});
+      EXPECT_THAT(metric_->Evaluate(argument + Δargument),
+                  RelativeErrorFrom(
+                      metric_->Evaluate(argument) +
+                          TransposedView(metric_->EvaluateGradient(argument)) *
+                              Δargument,
+                      AllOf(Ge(0.0), Le(0.0027))));
```

Comment:
Get the maximal relative error, move the expectation out of the loops, and expect it with an IsNear on a quantity passed in the instantiation.

Same for 🍰.